torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -628,6 +631,71 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
         return outputs
 
+    @torch.no_grad()
+    def _write_back_current_shard(self):
+        for p in self.params:
+            if not p._is_sharded:
+                pass
+            chunks = p._full_param_padded.chunk(self.world_size)
+            if len(chunks) < (self.rank + 1):
+                # if there are not enough chunks, there's nothing to writeback here
+                pass
+        else:
+            chunk = chunks[self.rank]
+            p._local_shard.copy_(chunk)
+
+    @contextlib.contextmanager
+    def _summon_full_params(self, recurse: bool = True, volatile: bool = False) -> Generator:
+        """
+        A context manager to expose full params for the current FSDP instance.
+        Can be useful *after* forward/backward for a model to get the params for
+        additional processing or checking.
+        .. note:: This can be used on inner FSDPs.
```

Comment:
For the sake of understanding, does this mean that if I have:
```
FSDP(
nn.Linear,
FSDP(nn.Linear),
)
```

if I use it on the nested FSDP, I'll get only the nested linear layer's parameters? Do we have the appropriate unittest for this?

Reply:
Yup, that's what it means. We don't have a test specifically for that, I'm adding a test case for that.